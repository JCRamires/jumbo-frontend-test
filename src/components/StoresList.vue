<script setup lang="ts">
import { JumboStore } from '../stores/jumboStores.js';
import StoreListItem from './StoreListItem.vue'

interface Props {
  stores: JumboStore[];
}

const { stores } = defineProps<Props>();
</script>

<template>
  <div class="grid md:grid-cols-[repeat(auto-fit,_minmax(25vw,_1fr))] gap-2">
    <StoreListItem v-for="store in stores" :store="store" :key="store.sapStoreID" />
  </div>
</template>
