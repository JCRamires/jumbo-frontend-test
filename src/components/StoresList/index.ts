import StoresList from "./StoresList.vue";

export default StoresList;
