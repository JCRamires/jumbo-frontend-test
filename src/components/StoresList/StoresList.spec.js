import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import StoresList from "./StoresList.vue";

const STORES = [
  {
    type: "StoreListRO",
    uuid: "V7cKYx4X0QUAAAFMTmYM5CXj",
    addressName: "Jumbo Alkmaar Duijvelshoff",
    street: "Muiderwaard",
    street2: "416",
    street3: "",
    city: "Alkmaar",
    postalCode: "1824 XT",
    distance: 0,
    todayOpen: "07:00",
    todayClose: "22:00",
    latitude: "52.645601",
    longitude: "4.749492",
    locationType: "SupermarktPuP",
    newStore: 0,
    collectionPoint: true,
    complexNumber: "33282",
    sapStoreID: "3574",
    favourite: false,
    isHomeStore: false,
    showWarningMessage: true,
    sundayOpen: true,
  },
  {
    type: "StoreListRO",
    uuid: "xSgKYx4XXk0AAAFTOqsG6pvj",
    addressName: "Jumbo Amersfoort Leusderweg",
    street: "Leusderweg",
    street2: "152",
    street3: "",
    city: "Amersfoort",
    postalCode: "3817 KD",
    distance: 0,
    todayOpen: "07:30",
    todayClose: "21:00",
    latitude: "52.143866",
    longitude: "5.381369",
    locationType: "SupermarktPuP",
    newStore: 0,
    collectionPoint: true,
    complexNumber: "32257",
    sapStoreID: "4730",
    favourite: false,
    isHomeStore: false,
    showWarningMessage: true,
    sundayOpen: true,
  },
];

describe("StoresList", () => {
  it("mounted component should match snapshot", async () => {
    const wrapper = mount(StoresList, {
      props: {
        stores: STORES,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
