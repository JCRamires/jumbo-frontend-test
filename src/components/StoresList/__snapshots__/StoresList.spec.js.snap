// Vitest Snapshot v1

exports[`StoresList > mounted component should match snapshot 1`] = `
<div
  class="grid md:grid-cols-[repeat(auto-fit,_minmax(25vw,_1fr))] gap-2"
>
  
  <router-link
    to="/stores/3574"
  >
    <div
      class="border rounded shadow p-4 h-full"
    >
      <div
        class="font-semibold mb-2"
      >
        Jumbo Alkmaar Duijvelshoff
      </div>
      <div
        class="flex gap-1 items-center"
      >
        <svg
          aria-hidden="true"
          class="h-5 w-5 text-gray-400"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.657 16.657L13.414 20.9a1.998 1.998 0 01-2.827 0l-4.244-4.243a8 8 0 1111.314 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
          <path
            d="M15 11a3 3 0 11-6 0 3 3 0 016 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <div>
          Muiderwaard, 416
        </div>
      </div>
      <div
        class="flex gap-1 items-center"
      >
        <svg
          aria-hidden="true"
          class="h-5 w-5 text-gray-400"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 21V5a2 2 0 00-2-2H7a2 2 0 00-2 2v16m14 0h2m-2 0h-5m-9 0H3m2 0h5M9 7h1m-1 4h1m4-4h1m-1 4h1m-5 10v-5a1 1 0 011-1h2a1 1 0 011 1v5m-4 0h4"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <div>
          Alkmaar
        </div>
      </div>
      <div
        class="flex gap-1 items-center"
      >
        <svg
          aria-hidden="true"
          class="h-5 w-5 text-gray-400"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <div>
          07:00 to 22:00
        </div>
      </div>
    </div>
  </router-link>
  <router-link
    to="/stores/4730"
  >
    <div
      class="border rounded shadow p-4 h-full"
    >
      <div
        class="font-semibold mb-2"
      >
        Jumbo Amersfoort Leusderweg
      </div>
      <div
        class="flex gap-1 items-center"
      >
        <svg
          aria-hidden="true"
          class="h-5 w-5 text-gray-400"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.657 16.657L13.414 20.9a1.998 1.998 0 01-2.827 0l-4.244-4.243a8 8 0 1111.314 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
          <path
            d="M15 11a3 3 0 11-6 0 3 3 0 016 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <div>
          Leusderweg, 152
        </div>
      </div>
      <div
        class="flex gap-1 items-center"
      >
        <svg
          aria-hidden="true"
          class="h-5 w-5 text-gray-400"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 21V5a2 2 0 00-2-2H7a2 2 0 00-2 2v16m14 0h2m-2 0h-5m-9 0H3m2 0h5M9 7h1m-1 4h1m4-4h1m-1 4h1m-5 10v-5a1 1 0 011-1h2a1 1 0 011 1v5m-4 0h4"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <div>
          Amersfoort
        </div>
      </div>
      <div
        class="flex gap-1 items-center"
      >
        <svg
          aria-hidden="true"
          class="h-5 w-5 text-gray-400"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <div>
          07:30 to 21:00
        </div>
      </div>
    </div>
  </router-link>
  
</div>
`;
